app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -0,0 +1,212 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_make_request);
+        setTitle("New Request");
+
+        activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
+
+        Button locationButton = (Button) findViewById(R.id.button_chooseLocation);
+        locationButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                setResult(RESULT_OK);
+                chooseLocations();
+            }
+        });
+
+        Button estimateButton = (Button) findViewById(R.id.button_estimateFare);
+        estimateButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                setResult(RESULT_OK);
+                estimateFare();
+            }
+        });
+
+        Button submitButton = (Button) findViewById(R.id.button_submit);
+        submitButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                setResult(RESULT_OK);
+                submitRequest();
+            }
+        });
+    }
+
+    private void chooseLocations() {
+        Toast.makeText(activity, "Choose locations", Toast.LENGTH_SHORT).show();
+
+        if(start == null) start = new Location();
+        if(end == null) end = new Location();
+
+        // TODO remove fake data
+        // fake data, University of Alberta
+        start.setLocation(53.5232, -113.5263);
+        // fake data, Edmonton City Centre
+        end.setLocation(53.5438, -113.4923);
+
+        // prompts for fake data verification
+        Toast.makeText(activity, "Location 1: (53.5232, -113.5263)", Toast.LENGTH_SHORT).show();
+        Toast.makeText(activity, "Location 2: (53.5438, -113.4923)", Toast.LENGTH_SHORT).show();
+
+        // TODO create the activity for selecting locations of request (later use case)
+        //    the new activity will handle everything and will return back to use the
+        //    start and end location coordinates
+        // SetLocationsActivity does not exist, placeholder name
+        //   this will be where the user sets the start and end locations of their request
+        // Intent intent = new Intent(MakeRequestActivity.this, SetLocationsActivity.class);
+        // startActivity(intent);
+    }
+
+    /**
+     * When back is pressed
+     *      Pop up an AlertDialog to confirm that the user will cancel their request.
+     *      Return to MainActivity.
+     */
+    public void onBackPressed() {
+        AlertDialog.Builder adb = new AlertDialog.Builder(this);
+        adb.setTitle("Are you sure?");
+        adb.setMessage("Cancel request?");
+        adb.setCancelable(true);
+        final Activity activity = MakeRequestActivity.this;
+        // TODO semantics re: cancel, button text (is it confusing to have "cancel" and "cancel request"?)
```

Comment:
Leave?
Are you sure? You will lose this request if you leave.
(Yes) (Stay | No | oh god )